src/main/java/de/wejago/hichi2influx/service/MqttSubscriberService.java

```diff
@@ -0,0 +1,58 @@
+package de.wejago.hichi2influx.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.wejago.hichi2influx.config.MqttClientConfig;
+import de.wejago.hichi2influx.repository.InfluxDbRepository;
+import jakarta.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.paho.client.mqttv3.IMqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class MqttSubscriberService {
+    @Value("${mqtt.topic}")
+    private String topic;
+
+    private final InfluxDbRepository influxDbRepository;
+
+    private final SensorMessageSubscriber sensorMessageSubscriber;
+
+    private final MqttClientConfig mqttClientConfig;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private IMqttClient mqttClient;
+
+    private MqttConnectOptions connectOptions;
+
+    @PostConstruct
+    public void postConstruct() {
+        connect();
+        subscribe();
+    }
+
+    public void connect() {
+        try {
+            mqttClient = mqttClientConfig.setMqttClient();
+            connectOptions = mqttClientConfig.setMqttConnectOptions();
+            mqttClient.connect(connectOptions);
+            log.info("Successfully connected to MQTT client!");
+        } catch (MqttException e) {
+            log.error("Error connecting to MQTT client!", e);
+        }
+    }
+
+    public void subscribe() {
```

Comment:
I think you can merge connect and subscribe, they belong together.
And then it can be the @PostConstruct method.
So the class gets leaner.